### PR TITLE
Refactor inventory plugin

### PR DESCRIPTION
**Pull request by fabianvf**
##### SUMMARY
Inventory plugin only adds valid hosts and groups
    
- No longer adds services to lists of hosts
- Group names are now sanitized
- Removed the `*_pods` groups, as pods are the only valid hosts anyway
- Add new groups for Deployments, Daemonsets and StatefulSets, allowing
  easy access to pods that are created by those resources
- Move extract_selector method to common

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME
plugins/inventory/k8s.py

##### ADDITIONAL INFORMATION
Technically modifying the group names to be valid could be considered a breaking change, though it's probably a bug that they weren't in the first place.

### CODE REVIEW

File: plugins/inventory/k8s.py
Comment (Akasurde) [inline plugins/inventory/k8s.py:212]: There is a function called [`_sanitize_group_name`](https://github.com/ansible/ansible/blob/48f12c14e9ff2b04c49cf6f90a65f4a9ecbb7fc7/lib/ansible/plugins/inventory/__init__.py#L153). Can we use this?

Comment (fabianvf) [reply plugins/inventory/k8s.py:212]: awesome, yeah we can definitely use that

**Comment by geerlingguy** [comment]
I like:

  * No longer adds services to lists of hosts
  * Removed the `*_pods` groups, as pods are the only valid hosts anyway

And I have never used those features (and have only tested the inventory plugin once anyways), but if we merge those changes, we will need to bump to a 2.0.0 version.

We could merge the rest of the changes in 1.x, I think—and then maybe you can move those other two changes to a PR that we could put in a 2.0.0 milestone?

**Comment by tima** [comment]
@fabianvf I agree about the group renaming and have to wonder if many at all where using the parts you touched. A breaking change is a breaking change that I'm going to attach this one to the 2.0 project.

File: plugins/inventory/k8s.py
Comment (geerlingguy) [reply plugins/inventory/k8s.py:231]: That I do not; you'd have to get @Akasurde's input, I have never built an inventory plugin (yet—I'm going to work on getting one merged for DigitalOcean soon).

Comment (Akasurde) [reply plugins/inventory/k8s.py:231]: We can use `strict` parameter and depend upon strictness (`strict: true`) we take action.

like -

```
if strict:
   raise AnsibleError("Could not...
else:
   display.warning("Could not...
```

You can find the example https://github.com/ansible-collections/vmware/blob/341c3375941d3119a61c10359c85a4caa8082e18/plugins/inventory/vmware_vm_inventory.py#L446 and https://github.com/ansible-collections/vmware/blob/341c3375941d3119a61c10359c85a4caa8082e18/plugins/inventory/vmware_vm_inventory.py#L807

File: plugins/inventory/k8s.py
Comment (maxamillion) [inline plugins/inventory/k8s.py:17]: Is this a change in functionality or is this a correction to the documentation?

Comment (geerlingguy) [reply plugins/inventory/k8s.py:17]: That's a change in functionality; this PR would limit the scope of the inventory plugin to something that's more practical in real-world use, but because it changes the behavior and removes a feature, that should probably be put into a 2.x.

**Comment by Akasurde** [comment]
recheck

File: plugins/module_utils/common.py
Comment (goneri) [inline plugins/module_utils/common.py:785]: The method don't use `self`, can we convert it in a function and add test-coverage?

**Review by goneri — APPROVED** [review]
Looks good to me, but I still need to test the code.

**Comment by tima** [comment]
I don't feel we have enough feedback on the validity or necessity of this plugin. I'd like to see us hold up on this until we get that for or against it's continued inclusion.

**Pull request closed without merging**
